src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -2144,7 +2144,18 @@ void LocalDerivationGoal::runChild()
                without file-write* allowed, access() incorrectly returns EPERM
              */
             sandboxProfile += "(allow file-read* file-write* process-exec\n";
+
+            // We create multiple allow lists, to avoid exceeding a limit in the darwin sandbox interpreter.
+            // See https://github.com/NixOS/nix/issues/4119
+            // We split our allow groups approximately at half the actual limit, 1 << 16
+            int breakpoint = sandboxProfile.length() + (1 << 14);
```

Comment:
```suggestion
            const int breakpoint = sandboxProfile.length() + (1 << 14);
```

unless I'm missing something, we shouldn't need to ever modify `breakpoint` ?